core/kt-fast-collections/src/test/kotlin/TestArrayList.kt

```diff
@@ -32,10 +334,51 @@ class ArrayListTest {
         assertEquals(3, arrayList.size)
 
         arrayList.clear()
-        
+
         assert(arrayList.isEmpty())
         assertEquals(0, arrayList.size)
     }
 
-    // TODO: add more tests
+    @Test
+    fun `iterator should iterate over all elements and not beyond`() {
+        val arrayList = MutableIntArrayList()
+        assertEquals(true, arrayList.add(1))
+        assertEquals(true, arrayList.add(2))
+        assertEquals(true, arrayList.add(3))
+
+        val iterator = arrayList.iterator()
+
+        assertEquals(1, iterator.next())
+        assertEquals(2, iterator.next())
+        assertEquals(3, iterator.next())
+        assert(!iterator.hasNext())
+    }
+
+    @Test
+    fun `lists with same elements and same order should be equal`() {
+        val arrayList1 = MutableIntArrayList()
+        assertEquals(true, arrayList1.add(1))
+        assertEquals(true, arrayList1.add(2))
+        assertEquals(true, arrayList1.add(3))
+        val arrayList2 = MutableIntArrayList()
+        assertEquals(true, arrayList2.add(1))
+        assertEquals(true, arrayList2.add(2))
+        assertEquals(true, arrayList2.add(3))
+
+        assertEquals(true, arrayList1 == arrayList2)
+    }
+
+    @Test
+    fun `lists with same elements and different order should be not equal`() {
+        val arrayList1 = MutableIntArrayList()
+        assertEquals(true, arrayList1.add(2))
+        assertEquals(true, arrayList1.add(3))
+        assertEquals(true, arrayList1.add(1))
+        val arrayList2 = MutableIntArrayList()
+        assertEquals(true, arrayList2.add(1))
+        assertEquals(true, arrayList2.add(2))
+        assertEquals(true, arrayList2.add(3))
+
+        assertEquals(false, arrayList1 == arrayList2)
```

Comment:
assertNotEquals may be useful https://kotlinlang.org/api/latest/kotlin.test/kotlin.test/